### ISSUE TITLE
`finally` blocks are not invoked if example fails

### ISSUE THREAD

**Issue by jonathanperret** [antonmi/espec#72]
Given the following spec:

```elixir
defmodule Spec.FailingSpec do
  use ESpec

  finally do
    IO.puts "in finally"
  end

  example "failing example" do
    expect 1 |> to(eq 2)
  end
end
```

When running the spec, I would expect the text `in finally` to appear on the output. RSpec `after` blocks are run regardless of example (or `before` block) result.

**Comment by antonmi** [comment]
Hi, Jonathan!
You are right. Will fix this soon.
Thanks!

**Comment by antonmi** [comment]
Fixed in 0.8.9.
